Sources/NIOSSL/TLSConfiguration.swift

```diff
@@ -446,7 +468,38 @@ extension TLSConfiguration {
                                 keyLogCallback: nil,
                                 renegotiationSupport: .none,
                                 additionalTrustRoots: [],
-                                sendCANameList: false)
+                                sendCANameList: false,
+                                pskClientCallback: nil,
+                                pskServerCallback: nil,
+                                pskHint: nil)
+    }
+    
+    /// Create a TLS configuration for use with server-side or client-side contexts that uses Pre-Shared Keys for TLS 1.2 and below.
+    ///
+    /// This provides sensible defaults while requiring that you provide any data that is necessary
+    /// for server-side or client-side functionality.  This configuration uses Pre-Shared Keys instead of certificates.
+    ///
+    /// For customising fields, modify the returned TLSConfiguration object.
+    public static func makePreSharedKeyConfiguration() -> TLSConfiguration {
+        
+        return TLSConfiguration(cipherSuites: defaultCipherSuites,
+                                verifySignatureAlgorithms: nil,
+                                signingSignatureAlgorithms: nil,
+                                minimumTLSVersion: .tlsv1,
+                                maximumTLSVersion: nil,
+                                certificateVerification: .none,
+                                trustRoots: .default,
+                                certificateChain: [],
+                                privateKey: nil,
+                                applicationProtocols: [],
+                                shutdownTimeout: .seconds(5),
+                                keyLogCallback: nil,
+                                renegotiationSupport: .none,
+                                additionalTrustRoots: [],
+                                sendCANameList: false,
+                                pskClientCallback: nil,
+                                pskServerCallback: nil,
+                                pskHint: nil)
     }
 }
```

Comment:
Can you extend `bestEffortEquals` and `bestEffortHash` here, and add appropriate testing? See also #386 (my fault for not catching that in review last time).

Reply:
Addressed in [f961f4a](https://github.com/apple/swift-nio-ssl/pull/369/commits/f961f4a6d34a6f5f40bc79b33e38e90ae76a12e3)